tasks/sbd.yml

```diff
@@ -0,0 +1,61 @@
+# SPDX-License-Identifier: MIT
+---
+# TODO: make creating SBD headers idempotent
+# - name: Check SBD device headers
+#   command: "sbd -d {{ item }}  dump"
+#   run_once: true
+#   ignore_errors: true
+#   register: "sbd_dump_{{ item }}"
+#   loop: "{{ ha_cluster_sbd_devices }}"
+#   when:
+#     - ha_cluster_use_sbd
+#     - ha_cluster_sbd_devices
+#   tags:
+#     - sbd
+
+- name: Create SBD devices
+  command: "sbd -d {{ item }} {{ ha_cluster_sbd_create_params }} create"
+  run_once: true
+  loop: "{{ ha_cluster_sbd_devices }}"
+  when:
+    - ha_cluster_use_sbd
+    - ha_cluster_sbd_devices
+  tags:
+    - sbd
+
+- name: Enable softdog module
+  copy:
+    content: "softdog"
+    dest: /etc/modules-load.d/softdog.conf
+    owner: root
+    group: root
+    mode: '0644'
+  notify: Load softdog module
+  tags:
+    - sbd
+
+- name: Set SBD parameters
+  replace:
+    path: /etc/sysconfig/sbd
+    regexp: "{{ item.regexp }}"
+    replace: "{{ item.replace }}"
+  loop:
+    - regexp: '^(.*)SBD_DEVICE=(.*)$'
+      replace: 'SBD_DEVICE="{{ ha_cluster_sbd_devices | join(";") }}"'
+    - regexp: '^(.*)SBD_OPTS=(.*)$'
+      replace: SBD_OPTS="{{ ha_cluster_sbd_daemon_params }}"
+    - regexp: '^(.*)SBD_STARTMODE=(.*)$' 
+      replace: "SBD_STARTMODE={{ ha_cluster_sbd_startmode }}"
```

Comment:
Is there a chance that some other software will be managing the contents of `/etc/sysconfig/sbd` e.g. do you want to handle the case where the user may have added something to `SBD_DEVICE`, `SBD_OPTS`, or `SBD_STARTMODE`, and you want to preserve it, in addition to the changes made by the ha_cluster role? If so, then I would suggest writing a module in order to handle preserving existing values.

If not, then I would suggest creating a jinja2 `template` for this.

Reply:
I wouldn't expect the user to alter the SBD configuration outside this role. It is mainly configured for cluster purposes. Also, the user has the option to set the most configured values using the role variables.
I will convert this into a jinja2 `template`.